tensorflow_probability/python/distributions/loglogistic.py

```diff
@@ -0,0 +1,175 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ============================================================================
+"""LogNormal distribution classes."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+import tensorflow.compat.v2 as tf
+
+from tensorflow_probability.python.bijectors import exp as exp_bijector
+from tensorflow_probability.python.distributions import logistic
+from tensorflow_probability.python.distributions import transformed_distribution
+from tensorflow_probability.python.internal import assert_util
+from tensorflow_probability.python.internal import dtype_util
+from tensorflow_probability.python.internal import tensor_util
+
+__all__ = [
+    'LogLogistic',
+]
+
+
+class LogLogistic(transformed_distribution.TransformedDistribution):
+  """The log-logistic distribution."""
+
+  def __init__(self,
+               scale,
+               concentration,
+               validate_args=False,
+               allow_nan_stats=True,
+               name='LogLogistic'):
+    """Construct a log-logistic distribution.
+
+    The LogLogistic distribution models positive-valued random variables
+    whose logarithm is logisticly distributed with mean `log(scale)` and
+    scale `1 / concentration`. It is constructed as the exponential
+    transformation of a Logistic distribution.
+
+    Args:
+      scale: Floating-point `Tensor`;
+        the scale of the log-Logistic distribution(s).
+      concentration: Floating-point `Tensor`;
+        the shape of the log- Logistic distribution(s).
+      validate_args: Python `bool`, default `False`. Whether to validate input
+        with asserts. If `validate_args` is `False`, and the inputs are
+        invalid, correct behavior is not guaranteed.
+      allow_nan_stats: Python `bool`, default `True`. If `False`, raise an
+        exception if a statistic (e.g. mean/mode/etc...) is undefined for any
+        batch member If `True`, batch members with valid parameters leading to
+        undefined statistics will return NaN for this statistic.
+      name: The name to give Ops created by the initializer.
+    """
+    parameters = dict(locals())
+    with tf.name_scope(name) as name:
+      dtype = dtype_util.common_dtype([scale, concentration],
+                                      dtype_hint=tf.float32)
+      self._scale = tensor_util.convert_nonref_to_tensor(
+          scale, name='scale', dtype=dtype)
+      self._concentration = tensor_util.convert_nonref_to_tensor(
+          concentration, name='concentration', dtype=dtype)
+      super(LogLogistic, self).__init__(
+          distribution=logistic.Logistic(loc=tf.math.log(self.scale),
```

Comment:
This can be problematic (this and below). This is because if these are `tf.Variable`, this won't be gradient tape safe.

I wonder if it's better to accept `loc` and `scale` as parameters. We do this for the `LogNormal` distribution and use the parametrization, and so would be consistent with it. WDYT?

Reply:
Why is it not gradient tape safe? Is it because the operations are defined at the initialisation ?

Regarding your suggestion, I'm not sure. I agree it would good to be consistent with the `LogNormal` but in the meantime, by doing so we lose the possibility to introduce a `loc` parameter later on  (and it makes me realise that if we keep this option I should remove:

```
  @property
  def loc(self):
    """Distribution parameter for the pre-transformed mean."""
    return self.distribution.loc
```

Reply:
Yep that's exactly it.

I would probably try to be consistent with LogNormal. Also IIRC, GeneralizedLogLogistic has a slightly different parameterization so we might want to have that as a separate implementation anyways.

Reply:
I see, thx.

I juste pushed a change but for some reasons I got nans for a couple of numpy tests only. Would you have any clue on why the numpy tests behave differently than the others?